python/Ganga/Core/GangaRepository/SessionLock.py

```diff
@@ -533,7 +533,7 @@ def session_read(self, fn):
             # This can fail (thats OK, file deleted in the meantime)
             fd = None
             try:
-                fd = os.open(fn, os.O_RDONLY)
+                fd = os.open(fn, os.O_RDWR)
```

Comment:
@rob-c Let us first see if this solves the problem for the user. My guess is that a write operation is reported back to the server (and thus can be seen by another client using the same file system). A read operation might stay local to the client. This is in fact a bit how afs works like.